src/main/java/org/deepsymmetry/beatlink/data/WaveformDetailComponent.java

```diff
@@ -940,7 +940,7 @@ public boolean isDynamicLoopDataAvailable() {
         if (!VirtualCdj.getInstance().isRunning()) return false;
         for (final PlaybackState state : playbackStateMap.values()) {
             final CdjStatus status = (CdjStatus) VirtualCdj.getInstance().getLatestStatusFor(state.player);
-            if (status.canReportLooping()) return true;
+            if (status != null && status.canReportLooping()) return true;
```

Comment:
`status` can be null because Opus will send empty updates when a song has not yet been loading into a player. We should skip in this case.

Reply:
This seems like it deserves to be a comment in the code itself.

Reply:
Good call, adding!